### PR TITLE
Fix symlinked venv ns-package calcs.

### LINKED ISSUE #2160: Bad `installed_wheels` dir when a wheel's `top_level.txt` dirname collides with `sources`
Steps to reproduce:

Here's a `pex` (renamed to `.zip` for GitHub's sake) [local.zip](https://github.com/pantsbuild/pex/files/11805798/local.zip). It has a `top_level.txt` of `top_level`. Rename to `local.whl`

Then, next to that file:
```console
$ mkdir -p sources/top_level/
$ echo 'print("hello")' > sources/top_level/mymain.py
$ pex --output-file binary.pex --venv prepend --requirements-pex local.pex --entry-point top_level.mymain $'--sources-directory=sources' --layout packed
$ ./binary/__main__.py
hello
$ ls ~/.pex/installed_wheels/a571cccbc564d1e7da4efe2416aca4bb049256ab4d0b30553f9d4de8207fb283/top_level-0.0.0-cp38-cp38-linux_x86_64.whl/top_level/
__init__.py  mymain.py  __pycache__
```

`mymain.py` is not part of the wheel, and shouldn't be in there.

### PR DESCRIPTION
**Pull request by jsirois**
Previously namespace packages were handled only amongst 3rd-party deps.
Any ns-package also split across PEX user sources would lead to one of
the 3rd-party dep members of the namespace package getting its cached
wheel install contaminated with the PEX user source that was a member of
the ns-package.

Fixes #2160

### CODE REVIEW

**Comment by jsirois** [comment]
The only commit to review is [0927f43](https://github.com/pantsbuild/pex/pull/2165/commits/0927f439f973beb7e8faac93197b1ebab27d1b0b) - the rest gets CI green and is off for review in #2162.

File: tests/integration/venv_ITs/test_issue_2160.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    )
+
+    distributions = list(PEX(pex).resolve())
+    assert 1 == len(distributions)
+
+    top_level = distributions[0]
+    assert ProjectName("top_level") == top_level.metadata.project_name
+    assert Version("0.1.0") == top_level.metadata.version
+
+    assert [os.path.join(top_level.location, "top_level", "lib.py")] == [
```
Comment (jsirois) [inline tests/integration/venv_ITs/test_issue_2160.py:118]: This test fails pre-fix for 2/9 cases as expected: symlinked venv {packed,zipapp}

**Review by benjyw — APPROVED** [review]
I only reviewed 0927f43, the rest looked like they're covered in other PRs.

File: tests/integration/venv_ITs/test_issue_2160.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+
+@pytest.fixture
+def top_level_wheel(tmpdir):
+    # type: (Any) -> str
+    top_level_project = os.path.join(str(tmpdir), "project")
+
+    top_level_library = os.path.join(top_level_project, "top_level", "lib.py")
+    with safe_open(top_level_library, "w") as fp:
+        fp.write("OG = 'Henry Barber'")
```
[the remark this answers is not in the document]
Comment (jsirois) [reply tests/integration/venv_ITs/test_issue_2160.py:31]: I grew up trying to climb Barber routes, sometimes succeeding. It's been a long time but just got on another one many years later out at the Ophir Wall. Even if not physically challenged it's almost certain he'll have provided a mental one!

**Pull request merged**